### ISSUE TITLE
idig_search not respecting limit param when creating attribution totals

### ISSUE THREAD

**Issue by kevinlove** [iDigBio/ridigbio#30]
Not sure if this is a bug or feature, but I'm getting some unexpected results from the dataframe that is returned by the idig_search method. 

```
genus <- "acer"
rq <-list("genus"=genus)
fields <- c('uuid', 'genus')
test_that("limited results, attribution counts match limit and result counts", {
        df <- idig_search_records(rq=rq, fields=fields, limit=10)
        expect_that(nrow(df) == 10, is_true())
        expect_that(attributes(df)$itemCount == 10, is_true())
})
```

It appears that the method respects the limit parameter, but reports back the totals from the API as an attribute of the dataframe. This seems appropriate, although unexpected. Lets look a little deeper into the results:

```
idig_build_attrib <- function(dat){
        dat <- attributes(dat)
        dx <- data.frame(stringsAsFactors = FALSE)
        for (i in (1:length(dat$attribution))){
                collection <- dat$attribution[[i]]$name
                uuid <- dat$attribution[[i]]$uuid
                itemCount <- dat$attribution[[i]]$itemCount
                rows <- cbind(collection,uuid,data.frame(itemCount,stringsAsFactors = FALSE))
                dx <- rbind(dx,rows)
        }
        return(dx)
}
genus <- "acer"
rq <-list("genus"=genus)
fields <- c('uuid', 'genus')        
df <- idig_search_records(rq=rq, fields=fields, limit=10)
dx <- idig_build_attrib(df)

test_that("limited results, attribution counts match result counts", {
        testthat::skip_on_cran()
expect_that(sum(dx$itemCount == 10), is_true())

})
```

So here we can see how it's going to be a difficult task to create a function for creating an attribution dataframe of the query results when the idig_search_records method doesn't respect the limit parameter. This issue seems like it should be solved upstream in the iDigBio API, but I wanted to see if there was any feedback from this group about the approach.

**Comment by mjcollin** [comment]
You're right, this is coming from the API, not from the ridigbio interface. The matching rows returned by the API relate to the the query being run, not the data being transferred. I'm having difficulty understanding the semantics of what a limited result set means in practice ie "the first 10 records of genus=acer" when the ordering of records in iDigBio as returned by the ridigbio package (not the API in general) is by UUID which is effectively random order.

If you're set on building a citation list only for specific records, then you can summarize the data frame by recordsetID and open a ticket on the API side to ask for a recordset endpoint, which last I know we didn't have yet, to iterate against.

**Comment by kevinlove** [comment]
> If you're set on building a citation list only for specific records, then you can summarize the data frame by recordsetID and open a ticket on the API side to ask for a recordset endpoint, which last I know we didn't have yet, to iterate against.

This is going to be difficult to achieve, unless the idig_search method includes recordsetID as a default field OR a user does not exclude this data, as my citation function works on the output of this method.

> I'm having difficulty understanding the semantics of what a limited result set means in practice

I'll open a ticket on the API side for this, but I feel that if I ask the API to limit a query, the resulting citation information that is returned should be constrained by that limit request, ie I ask for 10 records of a particular genus, just give me attribution information for those 10 records.

**Comment by mjcollin** [comment]
Adding the recordsetID to the default is not a bad idea in general. Alex pointed out that you don't actually have to take that id back to a recordset API endpoint; you already have all the record set information since it was in the original attribution block.

So, idig_build_attrib could become a more generic function who's purpose is to take a dataframe from idig_search_\* and build a dataframe that provide the attribution for only those records. That side steps my perspective that the attribution returned by the API should refer to the query, not the records. Now such a function could be used post-cleanup or some other client-side processing and I feel that is an appropriate time to summarize the attribution.

**Comment by kevinlove** [comment]
> idig_build_attrib could become a more generic function

That's exactly what I've been working on, but the way I've been approaching it has upstream implications on the way the data frame is assembled. Not only do you have to include "recordset" as a default field, you'll have to append it to the field vector supplied to the idig_search method...

**Comment by mjcollin** [comment]
Resolved by merging pull #32 & release 0.3.4
